src/Runner.Listener/JobDispatcher.cs

```diff
@@ -767,6 +771,25 @@ public async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockToke
             }
         }
 
+        private void UpdateAgentNameIfNeeded(string agentName)
+        {
+            try
+            {
+                if (_runnerSettings.AgentName != agentName)
+                {
+                    _runnerSettings.AgentName = agentName;
+                    _configurationStore.SaveSettings(_runnerSettings);
+                }
+
+            }
+            catch (IOException ex)
+            {
+                // Warning, not an Error, because the job execution continues
```

Comment:
nit: is this comment from copy paste?

Reply:
No, I forgot to clean it up - I used to log an exception with `Trace.Warning` and wanted to leave a comment to explain why.